Competition2017/tests/competition/subsystems/drive/commands/FieldOrientedTankDriveCommandTest.java

```diff
@@ -0,0 +1,76 @@
+package competition.subsystems.drive.commands;
+
+import org.junit.Before;
+import org.junit.Test;
+
+import competition.operator_interface.OperatorInterface;
+import competition.subsystems.drive.DriveTestBase;
+import xbot.common.controls.sensors.MockJoystick;
+
+public class FieldOrientedTankDriveCommandTest extends DriveTestBase {
+
+    private FieldOrientedTankDriveWithJoystick command;
+    private OperatorInterface oi;
+    
+    @Before
+    public void setUp() {
+        super.setUp();
+        command = injector.getInstance(FieldOrientedTankDriveWithJoystick.class);
+        oi = injector.getInstance(OperatorInterface.class);
+    }
+    
+    @Test
+    public void goingLeftFromStraight() {
+        setLeftJoystick(-1, 0);
+        command.initialize();
+        command.isFinished();
+        command.execute();
+        
+        verifyDriveArcingLeft(0.5);
+    }
+    
+    @Test
+    public void goingRightFromStraight() {
+        setLeftJoystick(1, 0);
+        command.initialize();
+        command.isFinished();
+        command.execute();
+        
+        verifyDriveArcingRight(0.5);
+    }
+    
+    @Test
+    public void goingStraightFromStraight() {
+        setLeftJoystick(0, 1);
+        command.initialize();
+        command.isFinished();
+        command.execute();
+        
+        verifyDrivePositive();
+    }
+    
+    @Test
+    public void goingBackLeftFromStraight() {
+        setLeftJoystick(-0.1, -1);
+        command.initialize();
+        command.isFinished();
+        command.execute();
+        
+        verifyDriveArcingLeft(0.5);
+    }
+    
+    @Test
+    public void goingBackRightFromStraight() {
+        setLeftJoystick(0.1, -1);
+        command.initialize();
+        command.isFinished();
+        command.execute();
+        
+        verifyDriveArcingRight(0.5);
+    }
+    
```

Comment:
You should have a test where you start facing ahead, turn left, then change the heading and verify your code stops trying to turn.